pkg/frontend/querymiddleware/codec.go

```diff
@@ -365,8 +369,9 @@ func (c prometheusCodec) decodeRangeQueryRequest(r *http.Request) (MetricsQueryR
 	var options Options
 	decodeOptions(r, &options)
 
+	stats := reqValues.Get("stats")
```

Comment:
there is already an HTTP header which requests stats for the `server-timing` response header, can we reuse that? 

https://github.com/grafana/mimir/blob/8d8a0a82dc8b65222d389e0dfe8c291ec7869edb/pkg/frontend/transport/handler.go#L46

Reply:
I'm not sure. The problem is that Prometheus Query Engine will Report per step stats under two conditions:
1. Feature flag is enabled
2. Request has a stats="all" query param.

Yes, we currently use MQE, but it might be the case that it's disabled in favour of Prometheus Query Engine, because of a bug or something, so that's why I had to implement what Prometheus wants.

Reply:
i didn't know prometheus supports this too. So is your plan to also return these stats in the client response too?

It's unfortunate both projects implemented the same. Can we at least make it less confusing to use? I'm thinking that we can treat the header and the query param would mean the same thing - include `Server-Timing` response and also return per-step statistics.

Reply:
in a DM we agreed to keep them as separate concepts because we always want prometheus stats for internal billing, but maybe not always want the header stats